src/clients/dispatcher/heartbeat/heartbeat-controller.ts

```diff
@@ -0,0 +1,40 @@
+import { Logger } from '@hatchet/util/logger';
+import { DispatcherClient as PbDispatcherClient } from '@hatchet/protoc/dispatcher';
+import { Worker } from 'worker_threads';
+import path from 'path';
+import { runThreaded } from '@hatchet/util/thread-helper';
+import { ClientConfig } from '../../hatchet-client';
+import { DispatcherClient } from '../dispatcher-client';
+
+export class Heartbeat {
+  config: ClientConfig;
+  client: PbDispatcherClient;
+  workerId: string;
+  logger: Logger;
+
+  heartbeatWorker: Worker | undefined;
+
+  constructor(client: DispatcherClient, workerId: string) {
+    this.config = client.config;
+    this.client = client.client;
+    this.workerId = workerId;
+    this.logger = new Logger(`HeartbeatController`, this.config.log_level);
+  }
+
+  async start() {
```

Comment:
Why are these `async`?

Reply:
the heartbeat is async

Reply:
It doesn't `await` anything and these don't seem to be awaited, what am I missing?

Reply:
>     await beat();

Reply:
not awaited because its a top level function